competition/parsers.py

```diff
@@ -63,22 +65,67 @@ def parse(self):
             )
         return levels
 
-    def create_problems(self,game):
+    def create_problems(self, game):
         levels = self.parse()
-        previous_level =None
-        for i,level in enumerate(levels):
+        previous_level = None
+        for i, level in enumerate(levels):
             new_level = Level.objects.create(
                 game=game,
                 order=i+1,
                 previous_level=previous_level
             )
             previous_level = new_level
-            for j,(problem,result) in enumerate(zip(level['problems'],level['results'])):
+            for j, (problem, result) in enumerate(zip(level['problems'], level['results'])):
                 Problem.objects.create(
                     level=new_level,
-                    text = problem,
-                    order = j+1,
+                    text=problem,
+                    order=j+1,
                     solution=result
                 )
 
+# TODO: Should be defined in another file
+def generate_password():
+    return ''.join(choice('abcdefghijklmnopqrstuvwxyz') for _ in range(8))
 
+
+class CompetitorsParser(UTF8Parser):
+    def parse(self):
+        text = super().parse()
+        competitors = []
+        for user_line in text.split('\n'):
+            if not user_line:
+                continue
+            firstname, lastname, school, grade, legal_representative = user_line.split(';')
+            competitors.append(
+                {
+                    'firstname': firstname,
+                    'lastname': lastname,
+                    'school': school,
+                    'grade': Grade.grade_from_number(int(grade)),
```

Comment:
Nebude lepsie v tom csvcku pouzit ten shortcut ... mi to pride istejsie ako sa spoliehat na tu magiu s idckami

Reply:
Bude, prerobim